src/sampleinfo.cpp

```diff
@@ -259,8 +259,10 @@ SampleInfo::get_genotype_from_coverage() const
         std::tie(index_of_max_likelihood, confidence, max_likelihood)
             = *index_and_confidence_and_max_likelihood_optional;
 
-        const bool satisfy_confidence_threshold
-            = confidence > genotyping_options->get_confidence_threshold();
+        const double threshold = genotyping_options->get_confidence_threshold();
+        const auto conf_equals_threshold = boost::math::relative_difference(confidence, threshold) < 1E-5;
```

Comment:
Just realised we already have implemented another way to check if doubles/floats are almost equal: https://github.com/mbhall88/pandora/blob/76c4cafef92015b204e5e5d350de3fc3c8f4ac51/include/Maths.h#L110-L118 . It uses what GoogleTest uses to check if doubles/floats are almost equal. I think we should discuss both methods and choose one. Source for the GoogleTest check can be found [here](https://github.com/google/googletest/blob/main/googletest/include/gtest/internal/gtest-internal.h#L334-L347). Boost float comparison docs [here](https://www.boost.org/doc/libs/1_81_0/libs/math/doc/html/math_toolkit/float_comparison.html) . If both methods are equivalent, I'd be slightly leaning towards the GoogleTest one, since it seems they carefully account for some edge cases (e.g. `nan`s), and how many units in the last place to tolerate when comparing doubles (tweaked to ordinary use on Intel CPUs, see [kMaxUlps](https://github.com/google/googletest/blob/ec25eea8f8237cf86c30703f59747e42f34b6f75/googletest/include/gtest/internal/gtest-internal.h#L274-L286)). However, I am going to help on predocs training now, and won't have time to continue reading or check this afterwards... I can discuss further tomorrow

Reply:
Oh whoops, didn't realise we had a function already. I'll just use that then.  
Yeah, boost also has lots of edge case handling (see code [here](https://www.boost.org/doc/libs/1_62_0/boost/math/special_functions/relative_difference.hpp)). I think they're both effectively influenced by Knuth and [this blog post](https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/), which I would caution against. It sucked me in and I lost a whole morning and am now very scared of float comparison 😆